File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/OutputMessageDTO.java
package com.szlazakm.chatserver.dtos;

import lombok.EqualsAndHashCode;
import lombok.Value;

import java.util.UUID;

@EqualsAndHashCode(callSuper = true)
@Value
public class OutputMessageDTO extends MessageDTO {

    String date;

    public OutputMessageDTO(final UUID id, final String from, final String to, final String text, final String date) {
        this.setId(id);
        this.setFrom(from);
        this.setTo(to);
        this.setText(text);
        this.date = date;
    }
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/OutputMessageDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class OutputMessageDTO(
    @JsonProperty("id")
    val id: UUID,
    @JsonProperty("from")
    val from: String,
    @JsonProperty("to")
    val to: String,
    @JsonProperty("text")
    val text: String,
    @JsonProperty("date")
    val date: String
)
